Lib/pdb.py

```diff
@@ -362,7 +362,8 @@ def displayhook(self, obj):
     def default(self, line):
         if line[:1] == '!': line = line[1:]
         locals = self.curframe_locals
-        globals = self.curframe.f_globals
+        ns = self.curframe.f_globals.copy()
```

Comment:
What it the code that is being debugged changes `globals()`? This would mean any changes to self.curframe.f_globals by the calling code is dropped, is it not?